hudi-flink/src/main/java/org/apache/hudi/sink/utils/PayloadCreation.java

```diff
@@ -76,6 +78,9 @@ public static PayloadCreation instance(Configuration conf) throws Exception {
       ValidationUtils.checkState(preCombineField != null);
       Comparable<?> orderingVal = (Comparable<?>) HoodieAvroUtils.getNestedFieldVal(record,
           preCombineField, false);
+      if(orderingVal instanceof LocalDateTime) {
+        orderingVal = ((LocalDateTime) orderingVal).atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();
```

Comment:
Why not just use a long if the `orderingVal` data type has no timezone.

Reply:
The data types returned from `HoodieAvroUtils.getNestedFieldVal` can be any valid Avro types including string, int, decimal, uuid, etc. How do you to check for timezone in these?

Reply:
Then what is the theory to use `ZoneId.systemDefault()` ?

Reply:
When converting epoch to `LocalDateTime`, a timezone needs to be specified, and we are using `ZoneId.systemDefault()` there. Hence when converting back from `LocalDateTime` to epoch, we are using the same.